mcodingbot/plugins/highlights.py

```diff
@@ -2,21 +2,45 @@
 
 import asyncio
 from collections import defaultdict
-from typing import NamedTuple
+from typing import TYPE_CHECKING, NamedTuple, cast
 
 import crescent
 import hikari
 from asyncpg import UniqueViolationError
 from floodgate import FixedMapping
+from toolbox.members import calculate_permissions
 
 from mcodingbot.config import CONFIG
 from mcodingbot.database.models import Highlight, User, UserHighlight
 from mcodingbot.utils import Context, Plugin
 
+if TYPE_CHECKING:
+    from mcodingbot.bot import Bot
+
 MAX_HIGHLIGHTS = 25
 MAX_HIGHLIGHT_LENGTH = 32
 
 
+async def has_permission(
+    bot: Bot, guild_id: int, user_id: int, channel_id: int
+) -> bool:
+    member = bot.cache.get_member(guild_id, user_id)
+    if not member:
+        return False
+
+    channel = bot.cache.get_guild_channel(channel_id)
+    if not channel:
+        # must be a thread
+        thread = bot.rest.fetch_channel(channel_id)
+        assert isinstance(
+            thread, (hikari.GuildThreadChannel, hikari.GuildNewsThread)
+        )
+        return await has_permission(bot, guild_id, user_id, thread.parent_id)
+
+    permissions = calculate_permissions(member, channel)
+    return hikari.Permissions.VIEW_CHANNEL in permissions
```

Comment:
Variable name doesn't add clarity imo
```suggestion
    return hikari.Permissions.VIEW_CHANNEL in calculate_permissions(member, channel)
```

Reply:
Alternative is ```py
return hikari.Permissions.VIEW_CHANNEL in calculate_permissions(
    member, channel
)
```

Reply:
still better. Also this is the final straw im upping the line length.

Reply:
resolving cause of trag1c's vote